sdk/core/core-http/src/policies/proxyPolicy.ts

```diff
@@ -12,23 +12,54 @@ import { ProxySettings } from "../serviceClient";
 import { WebResourceLike } from "../webResource";
 import { Constants } from "../util/constants";
 import { URLBuilder } from "../url";
+import { getEnvironmentValue } from "../util/utils";
+
+let noProxyList: any = null;
 
 function loadEnvironmentProxyValue(): string | undefined {
   if (!process) {
     return undefined;
   }
 
-  if (process.env[Constants.HTTPS_PROXY]) {
-    return process.env[Constants.HTTPS_PROXY];
-  } else if (process.env[Constants.HTTPS_PROXY.toLowerCase()]) {
-    return process.env[Constants.HTTPS_PROXY.toLowerCase()];
-  } else if (process.env[Constants.HTTP_PROXY]) {
-    return process.env[Constants.HTTP_PROXY];
-  } else if (process.env[Constants.HTTP_PROXY.toLowerCase()]) {
-    return process.env[Constants.HTTP_PROXY.toLowerCase()];
+  let httpsProxy =
+    getEnvironmentValue(Constants.HTTPS_PROXY) || getEnvironmentValue(Constants.ALL_PROXY);
+  let httpProxy =
+    getEnvironmentValue(Constants.HTTP_PROXY) || getEnvironmentValue(Constants.ALL_PROXY);
+
+  return httpsProxy || httpProxy;
+}
+
+// Check whether uri match noProxyList. If match, won't set proxy settings while sending request to the uri.
+function isBypassed(uri: string) {
```

Comment:
Should this be memoized? There's a lot of work happening for each request.

Reply:
The noproxy is stored in the global arrary "noProxyList". It would be initialized during the first call.

Reply:
I believe @bterlson is suggesting caching the result of `isBypassed()` for uri/host?  The cache size might be exploding for uris with varying query parameters, but host usually remains the same within a library so caching it should not be a problem.

Reply:
Right, since this is in a "hot path", reparsing the URL and re-mapping to the noProxyList for each request seems like a lot of work we don't have to do. To @jeremymeng's point, I suspect the number of hosts should be in practice limited to one, but maybe a handful at worst, I'm not too concerned with memory usage, but if we wanted to go with an LRU cache or something I wouldn't be opposed.

Reply:
Implemented Cache isBypassed() by url. Do you think it is necessary to use LRU cache? I find a [LRU-Cache](https://www.npmjs.com/package/lru-cache) package, is it possible to use here?